test/cpp-raw/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+project (cpp-raw)
+
+set (PROJECT_INCLUDE_DIR ${PROJECT_SOURCE_DIR})
+set (PROJECT_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+
+# Define build directory.
+set(CMAKE_BINARY_DIR ${CMAKE_SOURCE_DIR}/build)
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+
+# C++14 support
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# Define source files.
+set(PROJECT_SRCS
+${PROJECT_SOURCE_DIR}/catch.hpp
+${PROJECT_SOURCE_DIR}/error_info.cpp
+${PROJECT_SOURCE_DIR}/interface_params.cpp
+${PROJECT_SOURCE_DIR}/iunknown.cpp
+${PROJECT_SOURCE_DIR}/main.cpp
+${PROJECT_SOURCE_DIR}/primitive_tests.cpp
+${PROJECT_SOURCE_DIR}/result.cpp
+${PROJECT_SOURCE_DIR}/return_interfaces.cpp
+${PROJECT_SOURCE_DIR}/stateful.cpp
```

Comment:
> Why is catch.hpp here, but not things like os.h? I'm guessing this should be just the compiled sources, so catch.hpp should be omitted as it's include-only library.

Fixed.

> More importantly, does CMake support wildcards? I would love to have *.cpp here like we do in the VS project. As it is, I'm afraid we'll have failed builds because people (me) using other build systems (VS solution) forget to update the other (CMake) build scripts.

I would add the wildcards once I have finalized the initial support for gcc. Until all the basic infrastructure is in-place the ability  to leave some tests out is a good thing, imho.